src/core/socket.c

```diff
@@ -2087,6 +2153,11 @@ static void socket_enter_running(Socket *s, int cfd) {
                 cfd = -1; /* We passed ownership of the fd to the service now. Forget it here. */
                 s->n_connections++;
 
+                if (s->max_connections_per_source > 0 ) {
+                        socket_peer_ref(p);
+                        service->peer = p;
```

Comment:
this is prettier when written like this:

```c
service->peer = socket_peer_ref(p);
```

In particular, as you can then drop the {}...